xla/service/gpu/triton_tiling_propagation.cc

```diff
@@ -786,7 +786,9 @@ DimOrderMapOrError GetPropagatedDimOrdersForDimAlteringOp(
       for (const Fragment* subdim : dst_logical[dim_idx]) {
         dst_fragments_order.push_back(*subdim);
         src_to_dst[subdim] = dst_fragments_order.size() - 1;
-        dim_numbers_present_in_dst.insert(subdim->dst_dim_number());
+        if (subdim->full_count() > 1) {
```

Comment:
Why is this correct? Does the dimension not appear in `dst`, or is it just trivial? (`dim_numbers_present_in_dst` seems to indicate that this condition only makes sense if the former is true; if so, please document why. If not, then we need to think about how to change the variable names and/or add comments to make the expectation clear.)

Reply:
> Does the dimension not appear in dst, or is it just trivial?

It's trivial. Updated for clarity.